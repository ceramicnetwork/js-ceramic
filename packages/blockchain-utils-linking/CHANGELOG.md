# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.2.0](/compare/@ceramicnetwork/blockchain-utils-linking@1.2.0-rc.0...@ceramicnetwork/blockchain-utils-linking@1.2.0) (2021-09-16)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [1.2.0-rc.0](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.4-rc.0...@ceramicnetwork/blockchain-utils-linking@1.2.0-rc.0) (2021-09-13)


### Features

* caip10 ethereum+eos integration test ([#1668](https://github.com/ceramicnetwork/js-ceramic/issues/1668)) ([4267d8b](https://github.com/ceramicnetwork/js-ceramic/commit/4267d8b31fca38a163bb009198a49e1de22b2a58)), closes [#1678](https://github.com/ceramicnetwork/js-ceramic/issues/1678)





## [1.1.4-rc.0](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.3...@ceramicnetwork/blockchain-utils-linking@1.1.4-rc.0) (2021-09-08)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.3](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.3-rc.0...@ceramicnetwork/blockchain-utils-linking@1.1.3) (2021-09-08)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.3-rc.0](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.2...@ceramicnetwork/blockchain-utils-linking@1.1.3-rc.0) (2021-09-02)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.2](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.6...@ceramicnetwork/blockchain-utils-linking@1.1.2) (2021-08-25)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.2-rc.6](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.5...@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.6) (2021-08-24)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.2-rc.5](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.4...@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.5) (2021-08-23)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## 1.1.2-rc.4 (2021-08-23)


### Bug Fixes

* **ci:** remove private flag ([9974009](https://github.com/ceramicnetwork/js-ceramic/commit/9974009be69382f2a2caf59f4ff72bf6aa12491b))





## [1.1.2-rc.3](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.2...@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.3) (2021-08-22)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.2-rc.2](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.1...@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.2) (2021-08-22)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## 1.1.2-rc.1 (2021-08-22)


### Bug Fixes

* **ci:** remove flag from npm ci cmd ([b8ca310](https://github.com/ceramicnetwork/js-ceramic/commit/b8ca3102963096626a46a3c78c705da26e977021))





## [1.1.2-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@1.1.1...@ceramicnetwork/blockchain-utils-linking@1.1.2-rc.0) (2021-08-13)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.1.1](/compare/@ceramicnetwork/blockchain-utils-linking@1.1.0-rc.1...@ceramicnetwork/blockchain-utils-linking@1.1.1) (2021-08-11)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [1.1.0](/compare/@ceramicnetwork/blockchain-utils-linking@1.1.0-rc.1...@ceramicnetwork/blockchain-utils-linking@1.1.0) (2021-08-11)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [1.1.0-rc.1](/compare/@ceramicnetwork/blockchain-utils-linking@1.1.0-rc.0...@ceramicnetwork/blockchain-utils-linking@1.1.0-rc.1) (2021-08-03)


### Features

* **tezos-blockchain-link:** linking and validating DIDs to Tezos wallets (#1601) b40709e, closes #1601





# [1.1.0-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@1.0.3-rc.0...@ceramicnetwork/blockchain-utils-linking@1.1.0-rc.0) (2021-07-30)


### Features

* **tezos-blockchain-link:** initial support for linking and validating DIDs to Tezos wallets (#1555) baa6c35, closes #1555





## [1.0.3-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@1.0.2...@ceramicnetwork/blockchain-utils-linking@1.0.3-rc.0) (2021-07-16)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.0.2](/compare/@ceramicnetwork/blockchain-utils-linking@1.0.1...@ceramicnetwork/blockchain-utils-linking@1.0.2) (2021-05-25)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [1.0.1](/compare/@ceramicnetwork/blockchain-utils-linking@1.0.0...@ceramicnetwork/blockchain-utils-linking@1.0.1) (2021-05-13)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [1.0.0](/compare/@ceramicnetwork/blockchain-utils-linking@1.0.0-rc.2...@ceramicnetwork/blockchain-utils-linking@1.0.0) (2021-05-06)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [1.0.0-rc.2](/compare/@ceramicnetwork/blockchain-utils-linking@1.0.0-rc.1...@ceramicnetwork/blockchain-utils-linking@1.0.0-rc.2) (2021-04-20)


### Bug Fixes

* **blockchain-utils-linking:** pass through opts authProvider for eoaSignAccount (#1268) ae33805, closes #1268
* Fix tests by using node environment for jest (#1212) 0f04006, closes #1212


### Features

* **blockchain-utils-linking:** default link sign message 42a1d7e
* **tile-doctype:** Update Tile API (#1180) 90973ee, closes #1180





# [1.0.0-rc.1](/compare/@ceramicnetwork/blockchain-utils-linking@0.4.0-rc.1...@ceramicnetwork/blockchain-utils-linking@1.0.0-rc.1) (2021-04-19)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [0.4.0-rc.1](/compare/@ceramicnetwork/blockchain-utils-linking@0.3.0...@ceramicnetwork/blockchain-utils-linking@0.4.0-rc.1) (2021-04-19)


### Bug Fixes

* Fix tests by using node environment for jest (#1212) aff01c6, closes #1212





# [0.4.0-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.3.0...@ceramicnetwork/blockchain-utils-linking@0.4.0-rc.0) (2021-04-02)


### Features

* **tile-doctype:** Update Tile API 48f30e1





# [0.3.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0...@ceramicnetwork/blockchain-utils-linking@0.3.0) (2021-04-02)


### Bug Fixes

* **blockchain-utils-linking:** Pin version of @glif/local-managed-provider to unbreak test (#1183) b5ae20f, closes #1183
* **common, logger:** Clean up dependencies (#1164) 191ad31, closes #1164


### Features

* **blockchain-utils-linking:** Addition of Avalanche C-Chain Support (#1157) eb90f18, closes #1157





## [0.2.1-rc.3](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0...@ceramicnetwork/blockchain-utils-linking@0.2.1-rc.3) (2021-03-26)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [0.2.1-rc.2](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0...@ceramicnetwork/blockchain-utils-linking@0.2.1-rc.2) (2021-03-26)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [0.2.1-rc.1](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0...@ceramicnetwork/blockchain-utils-linking@0.2.1-rc.1) (2021-03-26)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [0.2.1-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0...@ceramicnetwork/blockchain-utils-linking@0.2.1-rc.0) (2021-03-25)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [0.2.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.5...@ceramicnetwork/blockchain-utils-linking@0.2.0) (2021-03-22)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [0.2.0-rc.5](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.4...@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.5) (2021-03-12)


### Features

* upgrade 3id did resolver (#1108) 24ef6d4, closes #1108





# [0.2.0-rc.4](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.2...@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.4) (2021-03-09)


### Bug Fixes

* **blockchain-utils-linking:** Pin @ethersproject/providers to previous version (#1094) 89fcebe, closes #1094





# [0.2.0-rc.3](https://github.com/ceramicnetwork/js-ceramic/compare/@ceramicnetwork/blockchain-utils-linking@0.1.0...@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.3) (2021-02-25)


### Features

* support near protocol ([#881](https://github.com/ceramicnetwork/js-ceramic/issues/881)) ([d50f36e](https://github.com/ceramicnetwork/js-ceramic/commit/d50f36e0ecc24d26ae90a7ee4bedbe1ee006f0f7)), closes [#880](https://github.com/ceramicnetwork/js-ceramic/issues/880)





# [0.2.0-rc.2](/compare/@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.0...@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.2) (2021-02-22)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





# [0.2.0-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.1.0...@ceramicnetwork/blockchain-utils-linking@0.2.0-rc.0) (2021-02-22)


### Features

* support near protocol (#881) d50f36e, closes #881 #880





# [0.1.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.0.6-rc.0...@ceramicnetwork/blockchain-utils-linking@0.1.0) (2021-02-04)


### Features

* **blockchain-utils-linking:** Export auth providers (#882) e7e7163, closes #882





## [0.0.6-rc.0](/compare/@ceramicnetwork/blockchain-utils-linking@0.0.5...@ceramicnetwork/blockchain-utils-linking@0.0.6-rc.0) (2021-01-29)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [0.0.5](/compare/@ceramicnetwork/blockchain-utils-linking@0.0.4...@ceramicnetwork/blockchain-utils-linking@0.0.5) (2021-01-21)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [0.0.4](/compare/@ceramicnetwork/blockchain-utils-linking@0.0.3...@ceramicnetwork/blockchain-utils-linking@0.0.4) (2021-01-13)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## [0.0.3](/compare/@ceramicnetwork/blockchain-utils-linking@0.0.2...@ceramicnetwork/blockchain-utils-linking@0.0.3) (2021-01-13)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking





## 0.0.2 (2021-01-13)

**Note:** Version bump only for package @ceramicnetwork/blockchain-utils-linking
