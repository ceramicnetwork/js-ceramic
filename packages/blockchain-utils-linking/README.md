# Utils for linking blockchain accounts

> This package contains utility functions to link blockchain accounts to a DID.

### Installation
```shell
npm install @ceramicnetwork/blockchain-utils-linking
```

## Contributing
We are happy to accept small and large contributions. Make sure to check out the [Ceramic specifications](https://github.com/ceramicnetwork/specs) for details of how the protocol works.

## License

Apache-2.0 or MIT
