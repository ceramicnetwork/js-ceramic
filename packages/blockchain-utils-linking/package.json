{
  "name": "@ceramicnetwork/blockchain-utils-linking",
  "version": "0.0.6-rc.0",
  "description": "Blockchain utils for linking blockchain accounts to DID",
  "keywords": [
    "ceramic",
    "caip10-link"
  ],
  "homepage": "https://ceramic.network",
  "license": "(Apache-2.0 OR MIT)",
  "directories": {
    "lib": "lib"
  },
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "scripts": {
    "test": "../../node_modules/.bin/jest",
    "build": "../../node_modules/.bin/tsc -p tsconfig.json",
    "prepublishOnly": "npm run build",
    "prebuild": "npm run clean",
    "lint": "../../node_modules/.bin/eslint ./src --ext .js,.jsx,.ts,.tsx",
    "clean": "rm -rf ./lib"
  },
  "dependencies": {
    "@stablelib/sha256": "^1.0.0",
    "caip": "^0.9.2",
    "uint8arrays": "^2.0.5"
  },
  "devDependencies": {
    "@glif/filecoin-address": "^1.1.0-beta.15",
    "@glif/local-managed-provider": "^1.1.0-beta.15",
    "@polkadot/api": "^3.0.1",
    "@polkadot/keyring": "^5.0.1",
    "@polkadot/types": "^3.0.1",
    "@polkadot/util": "^5.0.1",
    "@polkadot/util-crypto": "^5.2.3",
    "@smontero/eosio-local-provider": "^0.0.3",
    "@tendermint/sig": "^0.6.0",
    "@zondax/filecoin-signing-tools": "^0.13.0",
    "eth-sig-util": "^3.0.0",
    "ganache-core": "^2.13.1",
    "near-api-js": "^0.36.3"
  },
  "jest": {
    "testEnvironment": "jest-environment-uint8array",
    "transformIgnorePatterns": [
      "/node_modules(?!/did-jwt)/"
    ]
  },
  "gitHead": "34eeee25597b0a60def72906c26d3afd6230aaf1"
}
