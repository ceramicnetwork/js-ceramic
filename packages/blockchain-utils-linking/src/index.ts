export * from "./auth-provider";
export * as ethereum from "./ethereum";
export * from "./util";
export * as filecoin from "./filecoin";
export * as polkadot from "./polkadot";
export * as eosio from "./eosio";
