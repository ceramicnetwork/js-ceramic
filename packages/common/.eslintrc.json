{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint", "import"],
  "rules": {
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/ban-ts-ignore": "off",
    "@typescript-eslint/ban-ts-comment": "off",
    "import/no-unresolved": "error",
    "import/default": "off"
  },
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"]
    },
    "import/resolver": {
      "typescript": {
        "extensions": []
      },
      "node": true
    }
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:import/typescript"
  ]
}
