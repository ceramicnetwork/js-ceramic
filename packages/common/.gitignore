node_modules/
coverage/
lib/
src/generated/
