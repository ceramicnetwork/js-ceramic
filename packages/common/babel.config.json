{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "node": "16"
        },
        "useBuiltIns": false,
        "debug": false
      }
    ],
    "@babel/preset-typescript"
  ],
  "plugins": [
    [
      "@babel/plugin-transform-modules-commonjs",
      {
        "allowTopLevelThis": true
      }
    ],
    ["@babel/plugin-proposal-decorators", { "legacy": true }]
  ]
}
