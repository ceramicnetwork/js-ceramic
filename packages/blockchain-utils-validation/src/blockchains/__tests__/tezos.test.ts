import { jest } from '@jest/globals'
import { TezosAuthProvider, TezosProvider } from '@ceramicnetwork/blockchain-utils-linking'
import { InMemorySigner } from '@taquito/signer'
import { LinkProof } from '@ceramicnetwork/blockchain-utils-linking'
import { AccountId, ChainId } from 'caip'
import { normalizeAccountId } from '@ceramicnetwork/common'

const did = 'did:3:bafysdfwefwe'
const privateKey = 'p2sk2obfVMEuPUnadAConLWk7Tf4Dt3n4svSgJwrgpamRqJXvaYcg1'

type HttpResponse = Response

type IoTestCase = {
  testName?: string
  error?: any
  pubkeyObject?(publicKey: string): Promise<HttpResponse>
}

type ProofTestCase = {
  message: string
  proof: () => LinkProof
}

const ioTestCases: IoTestCase[] = [
  // Validation not possible
  // - public key hash never used the blockchain
  // - public key can't be found from non-existant wallet (on the blockchain)
  // - signature can't be verified
  {
    testName: 'unable to validate when wallet address has never interacted with the blockchain',
    error: new Error('Fetch response erroraa'),
  },

  // Validation not possible
  // - public key is not published
  // - signature can not be verified
  {
    testName: 'unable to validate when wallet address has not been published to the blockchain',
    async pubkeyObject(): Promise<HttpResponse> {
      const { Response } = await import('cross-fetch')
      return new Response(
        JSON.stringify({
          pubkey: undefined,
        })
      )
    },
  },

  // Able to test validation
  // - the public key is published/found on the blockchain
  // - the signature can be verified
  {
    async pubkeyObject(publicKey?: string): Promise<HttpResponse> {
      const { Response } = await import('cross-fetch')
      return new Response(
        JSON.stringify({
          pubkey: publicKey,
        })
      )
    },
  },
]

const proofTestCases: ProofTestCase[] = [
  // valid proof
  {
    message: 'valid proof',
    proof: () => validProof,
  },
  // invalid proof
  {
    message: 'invalid proof',
    proof: () => invalidSignatureProof,
  },
  // invalid chain reference
  {
    message: 'invalid chain reference',
    proof: () => invalidChainIdProof,
  },
]

let provider: TezosProvider
let publicKey: string
let validProof: LinkProof
let invalidSignatureProof: LinkProof
let invalidChainIdProof: LinkProof
let responseResult: () => Promise<any>

// cache Date.now() to restore it after all tests
const dateNow = Date.now

jest.unstable_mockModule('cross-fetch', () => {
  const originalModule = jest.requireActual('cross-fetch') as any
  return {
    ...originalModule,
    default: () => responseResult(),
  }
})

beforeAll(async () => {
  // Mock Date.now() to return a constant value
  Date.now = () => 666000

  // initialize mock tezos provider
  const signer = await InMemorySigner.fromSecretKey(privateKey)
  provider = {
    signer,
  }

  // create proof for did
  publicKey = await provider.signer.publicKey()
  const authProvider = new TezosAuthProvider(provider)
  validProof = await authProvider.createLink(did)

  invalidSignatureProof = { ...validProof, signature: 'invalid' }

  const accountId = normalizeAccountId(validProof.account)
  const chainId = new ChainId(accountId.chainId)
  const invalidAccount = new AccountId({
    address: accountId.address,
    chainId: {
      namespace: chainId.namespace,
      reference: 'some unsupported chain reference',
    },
  })
  invalidChainIdProof = { ...validProof, account: invalidAccount.toString() }
})

afterAll(() => {
  // restore Date.now()
  Date.now = dateNow
  jest.clearAllMocks()
})

describe('Blockchain: Tezos', () => {
  describe('validateLink', () => {
    // create test cases
    // run test cases
    for (const { testName, pubkeyObject, error } of ioTestCases) {
      for (const { message, proof } of proofTestCases) {
        test(testName || message, async () => {
          const { validateLink } = await import('../../index.js')
          // mock axios response or error
          if (pubkeyObject) {
            responseResult = async () => pubkeyObject(publicKey)
          }
          if (error) {
            responseResult = async () => Promise.reject(error)
          }
          // wait for test with the proof from the test case
          await expect(validateLink(proof())).resolves.toMatchSnapshot()
        })
      }
    }
  })
})
