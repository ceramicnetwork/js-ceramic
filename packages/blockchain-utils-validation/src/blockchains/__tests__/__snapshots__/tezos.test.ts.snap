// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockchain: Tezos validateLink invalid chain reference 1`] = `null`;

exports[`Blockchain: Tezos validateLink invalid proof 1`] = `null`;

exports[`Blockchain: Tezos validateLink unable to validate when wallet address has never interacted with the blockchain 1`] = `null`;

exports[`Blockchain: Tezos validateLink unable to validate when wallet address has never interacted with the blockchain 2`] = `null`;

exports[`Blockchain: Tezos validateLink unable to validate when wallet address has never interacted with the blockchain 3`] = `null`;

exports[`Blockchain: Tezos validateLink unable to validate when wallet address has not been published to the blockchain 1`] = `null`;

exports[`Blockchain: Tezos validateLink unable to validate when wallet address has not been published to the blockchain 2`] = `null`;

exports[`Blockchain: Tezos validateLink unable to validate when wallet address has not been published to the blockchain 3`] = `null`;

exports[`Blockchain: Tezos validateLink valid proof 1`] = `
Object {
  "account": "tezos:NetXdQprcVkpaWU:tz3Lfm6CyfSTZ7EgMckptZZGiPxzs9GK59At",
  "did": "did:3:bafysdfwefwe",
  "message": "Link this account to your identity

did:3:bafysdfwefwe 
Timestamp: 666",
  "signature": "p2sigNYcqkNERyCnBb7PKwmuZqeXBZx4m2c8YKcR282iWN9iafSdcTrJ4VxQdANg1oo7nGteZ7uGEDiy1ZpAqCCb71xdy9MptM",
  "timestamp": 666,
  "version": 2,
}
`;
